types/backbone/index.d.ts

```diff
@@ -568,26 +568,28 @@ declare namespace Backbone {
 
         model: TModel;
         collection: Collection<TModel>;
-        // template: (json, options?) => string;
-        setElement(element: HTMLElement | JQuery, delegate?: boolean): View<TModel>;
-        setElement(element: any): View<TModel>;
-        id: string;
+        setElement(element: HTMLElement | JQuery): this;
+        id?: string;
         cid: string;
-        className: string;
+        className?: string;
         tagName: string;
 
-        el: any;
+        el: HTMLElement;
         $el: JQuery;
-        attributes: any;
+        attributes: { [id: string]: any };
```

Comment:
What advantage does this have over `any`?

Reply:
Attributes are set to HTMLElement, they must be a hash with string properties. I just made this part consistent with `VIewOptions` `attributes` type, that are set directly to `View`.

Reply:
I'm not sure I follow.  It sounds like you might be trying to prevent some values from being assigned to `attributes`, but it's not obvious (to me, at least) which values this type rules out.  Can you please provide an example?

Reply:
For example, `string` shouldn't be passed as `attributes` value. attributes used here https://github.com/jashkenas/backbone/blob/master/backbone.js#L1445 and jQuery expects PlainObject, not a string or something else.

Reply:
Sounds like you might want `{}`.

Reply:
I'm agree with you, but it looks like better to use `Record` instead of `object`. https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/ban-types.md#default-options

Reply:
The TS team disagrees with that guidance, but it doesn't make much difference either way.